db/db_impl/db_impl.h

```diff
@@ -2707,6 +2722,11 @@ class DBImpl : public DB {
 
   std::deque<ManualCompactionState*> manual_compaction_dequeue_;
 
+  // A set of all the directories used by this DB.
+  // Note that these paths do not have a tailing path separator "/"
+  // (kFilePathSeparator).
+  std::unordered_set<std::string> all_db_paths_;
```

Comment:
My sense is that it would be better to use an ordered set here, for predictability of behavior. For example I believe there are some cases where the error returned on a multiply-corrupt DB depends on the iteration order of this set. It's less confusing if that error is not dependent on factors like how RocksDB was compiled. This could also appear as flaky test behavior in the future.

Data structure performance should not really be a concern here, as these data structure ops are generally paired with IO ops.

Reply:
Thanks for catching this change that may cause regression.